### PR TITLE
Add notes for various JSON APIs

### PR DESCRIPTION
**Pull request by terrajobst**
@ahsonkhan

### CODE REVIEW

File: 2019/System.Text.Json-Misc/README.md [hunk 1]
```diff
@@ -0,0 +1,104 @@
+# Miscellaneous JSON items
+
+## `JsonStringEnumConverter`
+
+Approved |
+[Video](https://youtu.be/1k-niYNMo4w?list=PL1rZQsJPBU2S49OQPjupSJF-qeIEz9_ju&t=400)
```
Comment (ahsonkhan) [inline 2019/System.Text.Json-Misc/README.md]: Nice timestamps for the video :)

File: 2019/System.Text.Json-Misc/README.md [hunk 2]
```diff
@@ -0,0 +1,104 @@
...
+        public JsonStringEnumConverter(JsonNamingPolicy namingPolicy = null, bool allowIntegerValues = true);
+    }
+}
+```
+
+**Comments:**
+
+* It only supports case differences, but not additional characters like kebab
+  cases, i.e. we can't map the strings `some-enum-value` to `SomeEnumValue`. In
+  that case we throw a `JsonException` with an appropriate error cases.
```
Comment (ahsonkhan) [inline 2019/System.Text.Json-Misc/README.md]:
```suggestion
  that case we throw a `JsonException` with an appropriate error message.
```

File: 2019/System.Text.Json-Misc/README.md [hunk 3]
```diff
@@ -0,0 +1,104 @@
...
+* We match using `Enum.Parse()`. We first try case-sensitive matching and then
+  fall back to case-insensitive. This is not exactly how deal with casing for
```
Comment (ahsonkhan) [inline 2019/System.Text.Json-Misc/README.md]:
```suggestion
  fall back to case-insensitive. This is not exactly how we deal with casing for
```

File: 2019/System.Text.Json-Misc/README.md [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+
+        [EditorBrowsable(EditorBrowsableState.Never)]
+        public virtual int FindFirstByteToEncodeUtf8(ReadOnlySpan<byte> utf8Text);
+    }
+}
+```
+
+**Review:**
+
+* Looks good as proposed.
```
Comment (ahsonkhan) [inline 2019/System.Text.Json-Misc/README.md]: Do you have the list of GitHub issues we looked at outside of these 3 APIs? Maybe add them as a list to this doc, for reference.

Comment (terrajobst) [reply 2019/System.Text.Json-Misc/README.md]: I don't :-(

**Pull request merged**